src/plots/cartesian/select.js

```diff
@@ -226,10 +259,31 @@ module.exports = function prepSelect(e, startX, startY, dragOptions, mode) {
             else {
                 dragOptions.gd.emit('plotly_selected', eventData);
             }
+
+            if(currentPolygon && dragOptions.polygons) {
+                // save last polygons
+                dragOptions.polygons.push(currentPolygon);
+
+                // we have to keep reference to arrays container
+                dragOptions.mergedPolygons.length = 0;
+                [].push.apply(dragOptions.mergedPolygons, mergedPolygons);
+            }
         });
     };
 };
 
+function joinPolygons(list, poly) {
```

Comment:
Joining the polygons sounds very reasonable to me. Some users might expect that selections within selections would remove the inner polygons. Perhaps we'll need a layout options (e.g. `layout.multiselectionmode: 'join' | 'cut'`).

Any thoughts about this @alexcjohnson @chriddyp @cldougl @monfera ?

Reply:
so `'join'` mode would always add points the the selection (and do nothing if a selected point is selected a second time) and `'cut'` mode would always remove points from the selection (and do nothing if the point was not previously selected)?

Reply:
+1 for a `cut` mode (after seeing the join in action)

Reply:
My gut reaction: if you start a new selection *inside* an existing one, it's purely subtractive, if you start *outside* the existing ones it's purely additive. Would have to play with it implemented to see if this really feels intuitive though...

Reply:
Ok, added subtract selection by holding <kbd>Alt</kbd> in e8c18

Reply:
> My gut reaction: if you start a new selection inside an existing one, it's purely subtractive, if you start outside the existing ones it's purely additive.

+1

Reply:
> Ok, added subtract selection by holding <kbd>Alt</kbd> in e8c18f0

That *is* more powerful (as well as easier to code 😅 ), my only concern is whether people will be able to find it. Though I guess if they can find add they can find subtract...

(and I never saw `<kbd>` before, nice! 🎉 )

Reply:
Nice solution @dfcreative . The only (potential) problem I can think of is mobile and tablets users complaining no having access to Shift and Alt keys.

Reply:
@etpinard good point, we can use @alexcjohnson strategy for that. But I'd wait for persistent selections PR merged first